prompt/prompt-executor/prompt-executor-clients/prompt-executor-bedrock-client/src/jvmMain/kotlin/ai/koog/prompt/executor/clients/bedrock/modelfamilies/anthropic/BedrockAnthropicClaudeSerialization.kt

```diff
@@ -55,19 +53,32 @@ internal object BedrockAnthropicClaudeSerialization {
                 is Message.Assistant -> {
                     if (msg.content.isNotEmpty()) {
                         messages.add(
-                            BedrockAnthropicInvokeModelMessage(
-                                role = "assistant",
+                            BedrockAnthropicInvokeModelMessage.Assistant(
                                 content = listOf(BedrockAnthropicInvokeModelContent.Text(text = msg.content))
                             )
                         )
                     }
                 }
 
+                is Message.Reasoning -> {
+                    if (msg.content.isNotEmpty()) {
+                        messages.add(
+                            BedrockAnthropicInvokeModelMessage.Assistant(
+                                content = listOf(
+                                    BedrockAnthropicInvokeModelContent.Thinking(
+                                        signature = msg.encrypted!!,
```

Comment:
nit: It might be a bit better to throw a more descriptive exception explicitly in this case, instead of using double-bang that would produce generic NPE without additional context.